contracts/TokenVesting.sol

```diff
@@ -0,0 +1,140 @@
+        require(vestingStart_ != 0, "vestingStart cannot be initialized as 0");
+        solace = solace_;
+        vestingStart = vestingStart_;
+        cliff = vestingStart_ + 15768000; // Cliff is 6-months after vesting start.
+        vestingEnd = vestingStart_ + 94608000; // Vesting ends 3-years after vesting start.
+    }
+
+    /***************************************
+    INVESTOR FUNCTIONS
+    ***************************************/
+
+    /**
+     * @notice Function for investor to claim SOLACE tokens - will claim all redeemable tokens.
+     */
+    function claimTokens () external override nonReentrant {
+        require(totalInvestorTokens[msg.sender] != 0, "You have no tokens to claim");
+        uint256 redeemableUnlockedTokens = getRedeemableUnlockedTokens(msg.sender);
+        require(redeemableUnlockedTokens > 0, "You cannot claim any tokens at the moment"); // Placed require statement here so that getRedeemableUnlockedTokens(msg.sender) doesn't have to be computed twice
+        redeemedInvestorTokens[msg.sender] += redeemableUnlockedTokens;
+        SafeERC20.safeTransfer(IERC20(solace), msg.sender, redeemableUnlockedTokens);
+    }
+
+    /**
+     * @notice Calculates the amount of unlocked SOLACE tokens an investor can claim.
+     * @param investor Investor address.
+     * @return redeemableUnlockedAmount The amount of unlocked tokens an investor can claim from the smart contract.
+     */
+    function getRedeemableUnlockedTokens(address investor) public view override returns (uint256 redeemableUnlockedAmount) {
+        uint256 timestamp = block.timestamp;
+        uint256 redeemableUnlockedAmount;
+        if(timestamp <= cliff) {
+            return 0;
+        } else if(timestamp <= vestingEnd) {
+            uint256 totalUnlockedAmount = ( totalInvestorTokens[investor] * (timestamp - cliff) / (vestingEnd - cliff) );
+            redeemableUnlockedAmount = totalUnlockedAmount - redeemedInvestorTokens[investor];
+            return redeemableUnlockedAmount;
+        } else {
+            redeemableUnlockedAmount = totalInvestorTokens[investor] - redeemedInvestorTokens[investor];
+            return redeemableUnlockedAmount;
+        }
+    }
+
+    /***************************************
+    GOVERNANCE FUNCTIONS
+    ***************************************/
+
+    /**
+     * @notice Rescues excess [**SOLACE**](./SOLACE).
+     * Can only be called by the current [**governor**](/docs/protocol/governance).
+     * @dev Trusting governance to perform accurate accounting off-chain and ensure there is sufficient SOLACE in contract to make payouts as dictated in totalInvestorTokens mapping.
+     * @param amount Amount to send.
+     * @param recipient Address to send rescued SOLACE tokens to.
+     */
+    function rescueSOLACEtokens(uint256 amount, address recipient) external override onlyGovernance {
+        require(recipient != address(0x0), "zero address recipient");
```

Comment:
Red flag in here. Imo Vesting contract should be immutable as possible as. No in/out token mechanism even with "governance".

Reply:
agreed its a security issue but its possible to over send SOLACE to this contract and we should have a way to get them back. we can lock governance after we complete the transfer of SOLACE in